### PR TITLE
[Media Common]Fix the user setting memory free

### PR DESCRIPTION
**Pull request by pingliwa**
This commit fix the #1300

Signed-off-by: Wang, Pingli <pingli.wang@intel.com>

### CODE REVIEW

**Review by XinfengZhang — APPROVED** [review]
this mutex maybe removed in future after registry refined.

File: media_softlet/linux/common/os/mos_utilities_specific_next.cpp [hunk 1]
```diff
@@ -612,6 +613,7 @@ MOS_STATUS MosUtilitiesSpecificNext::UserFeatureSet(MOS_PUF_KEYLIST *pKeyList, M
     MOS_AtomicIncrement(&MosUtilities::m_mosMemAllocFakeCounter);  //ulValueBuf does not count it, because it is freed after the MEMNJA final report.
     MOS_OS_NORMALMESSAGE("ulValueBuf %p for key %s", ulValueBuf, NewKey.pValueArray[0].pcValueName);
 
+    m_userSettingMutex.Lock();
```
Comment (XinfengZhang) [inline media_softlet/linux/common/os/mos_utilities_specific_next.cpp:616]: how about multiple-process case?

**Comment by pingliwa** [comment]
Hi,
Different process loads different .so. The mutex is different.  This function only writes value to cpu memory not write to files.

Best Wishes,
Pingli

**Comment by hanlong1** [comment]
@pingliwa could you please rebase this PR?

**Comment by pingliwa** [comment]
> @pingliwa could you please rebase this PR?

Done

**Pull request merged**